### DISCUSSION TITLE
sync_worker_exec: Network:\n\tsync_worker_exec_payload: EOF

### DISCUSSION THREAD

**Discussion by Warxcell** [roadrunner-server/roadrunner#1238] [Q&A] [answered]
```
2022-07-31T06:03:32.498Z	WARN	server      	network error, worker will be restarted	{"reason": "network", "pid": 67, "internal_event_name": "EventWorkerError", "error": "sync_worker_exec: Network:\n\tsync_worker_exec_payload: EOF"}
2022-07-31T06:03:32.515Z	DEBUG	server      	worker stopped	{"internal_event_name": "EventWorkerWaitExit", "error": "signal: killed; process_wait: signal: killed", "errorCauses": [{"error": "signal: killed"}, {"error": "process_wait: signal: killed"}]}
```

Any ideas on that? I cannot really find anything useful.

Tried pipes/tcp/socket - happens everywhere.

**Comment by rustatian** [comment]
Hey @Warxcell 👋🏻 
I guess you did smt wrong with the worker. Could you please try this simple worker in your env: https://github.com/roadrunner-server/rr-e2e-tests/blob/master/php_test_files/psr-worker-bench.php ?

**Reply by Warxcell** [reply]
Worker works fine with few (like 30-40) requests and then suddenly it fails with that error.

**Reply by rustatian** [reply]
Could you please show RR's configuration?

**Reply by Warxcell** [reply]
```
version: "2.7"

server:
    command: "php /app/server.php"
#    relay: "tcp://127.0.0.1:6002"
#    relay: "unix:///tmp/rr.sock"

http:
    address: ${HOST}:${PORT}
    pool:
        debug: ${RR_DEBUG}
        num_workers: ${RR_WORKERS}
        max_jobs: 0
        allocate_timeout: 5m
    middleware: [ "http_metrics", "static", "sendfile", "gzip" ]
    # Settings for "static" middleware (docs: https://roadrunner.dev/docs/http-static).
    static:
        dir: "/app/public/"
        forbid: [ ".php" ]
        calculate_etag: false
        weak: false

rpc:
    listen: tcp://127.0.0.1:6001

logs:
    mode: ${RR_LOGS}
    output: stderr

metrics:
    address: 0.0.0.0:9253

status:
    address: 0.0.0.0:2114
```

**Reply by rustatian** [reply]
Is `pool.debug` set to true?

**Reply by Warxcell** [reply]
In this particular case RR_DEBUG = false, RR_WORKERS = 1

**Reply by rustatian** [reply]
What is your env? OS, RR version, PHP version.

**Reply by Warxcell** [reply]
I think I found the issue. In the code on particular controller there was `ini_set('max_execution_time', 60)` and `display_errors = Off`.
I removed the `ini_set` and set `display_errors = stderr` so eventual error would go to STDERR, rather than hidden. (probably it doesn't throw exception??) 

Seems fine now. But was very odd to debug, no logs or whatsoever. :)

**Reply by rustatian** [reply]
Hm, and what was the error in the stderr?

**Reply by Warxcell** [reply]
Before I put `display_errors = Off` - there was nothing in the log.
Now I see whole symfony error page in stderr (weird tho, I will investigate further).

This is how my workers looks like

```
$kernel = new Kernel($context['APP_ENV'], (bool)$context['APP_DEBUG']);

    return new class($kernel, new Psr17Factory()) implements RunnerInterface {
        public function __construct(
            private readonly Kernel $kernel,
            private readonly Psr17Factory $psrFactory
        ) {
        }

        public function run(): int
        {
            $this->kernel->boot();

            $worker = \Spiral\RoadRunner\Worker::create();
            $worker = new PSR7Worker($worker, $this->psrFactory, $this->psrFactory, $this->psrFactory);

            $handler = $this->kernel->getContainer()->get(AppHandler::class);

            // We need to close connections because of App\Core\Security\Role\RoleHierarchy shit which fetches roles on init.
            $doctrineHandler = $this->kernel->getContainer()->get(DoctrineHandler::class);
            $doctrineHandler->reset();

            while ($request = $worker->waitRequest()) {
                try {
                    $worker->respond($handler->handle($request));
                } catch (Throwable $e) {
                    $worker->getWorker()->error((string)$e);

                    // In case we want custom 500 page.
                    //$response = $this->psrFactory->createResponse(500);
                    //$response->getBody()->write((string)$e);
                    //$worker->respond($response);
                } finally {
                    gc_collect_cycles();
                }
            }

            return 0;
        }
    };
```

**Reply by rustatian** [reply]
Thanks, I'll ask our PHP team for the additional info about how to debug such things and will add your answer to the our error codes docs page: https://roadrunner.dev/docs/known-issues-stdout-crc/2.x/en (like this one)

**Reply by rustatian** [reply]
Could you please write your answer (first and second part) not as the reply, but as a separate message, so, I'll be able to mark this as answer from you? 👍🏻

**Comment by Warxcell** [comment, marked as answer]
In the code on particular controller there was `ini_set('max_execution_time', 60)` and `display_errors = Off`.
I guess max_execution_time is the reason why worker exits, and `display_errors = Off` is the cause for no logs or whatsoever.

I removed the ini_set and set display_errors = stderr so eventual error would go to STDERR, rather than hidden.
